kpops/api/__init__.py

```diff
@@ -0,0 +1,272 @@
+from __future__ import annotations
+
+import asyncio
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import kpops
+from kpops.api.logs import log, log_action
+from kpops.api.options import FilterType
+from kpops.api.registry import Registry
+from kpops.component_handlers import ComponentHandlers
+from kpops.component_handlers.kafka_connect.kafka_connect_handler import (
+    KafkaConnectHandler,
+)
+from kpops.component_handlers.schema_handler.schema_handler import SchemaHandler
+from kpops.component_handlers.topic.handler import TopicHandler
+from kpops.component_handlers.topic.proxy_wrapper import ProxyWrapper
+from kpops.config import KpopsConfig
+from kpops.pipeline import (
+    Pipeline,
+    PipelineGenerator,
+)
+from kpops.utils.cli_commands import init_project
+
+if TYPE_CHECKING:
+    from kpops.components import PipelineComponent
+    from kpops.components.base_components.models.resource import Resource
+    from kpops.config import KpopsConfig
+
+
+def parse_steps(steps: str) -> set[str]:
+    return set(steps.split(","))
+
+
+def generate(
+    pipeline_path: Path,
+    dotenv: list[Path] | None = None,
+    config: Path = Path(),
+    steps: str | None = None,
```

Comment:
for the Python API function signature I would prefer directly using `set[str] | None`, `parse_steps` should be moved to the CLI